src/App.jsx

```diff
@@ -1,65 +1,136 @@
+  };
+  const searchIconsByName = (name, theme, option) => {
+    const iconDivs = [];
+    // eslint-disable-next-line no-plusplus
+    for (let i = 0; i < EOSIconsList[option].length; i++) {
+      const icon = EOSIconsList[option][i];
+      const isFilledAvailable = "hasOutlined" in icon && icon.hasOutlined;
+      const isFilledSelected = theme === "Filled";
+      if (
+        icon.name.indexOf(name) !== -1 &&
+        (!isFilledSelected || (isFilledSelected && isFilledAvailable))
+      ) {
+        const nameIcon = `Eos${icon.name}${
+          isFilledSelected ? "Filled" : "Outlined"
+        }`;
+        const EOSReactIcon = EOSIcons[nameIcon];
+        if (EOSReactIcon === undefined) {
+          // eslint-disable-next-line no-continue
+          continue;
+        }
+        iconDivs.push(
+          <div
+            className="image-container"
+            key={nameIcon}
+            onClick={() => {
+              copyToClipboard(EOSReactIcon({ size: "xxxl" }));
+            }}
+          >
+            {EOSReactIcon({ size: "xl" })}
+          </div>
+        );
+      }
+    }
+    if (iconDivs.length === 0) {
+      return null;
+    }
+    return (
+      <div key={option}>
+        <div className="category-container">
+          <span>{option}</span>
+        </div>
+        <div className="image-box">{iconDivs}</div>
+      </div>
     );
-    request.onload = () => {
-      if (!request.response.includes("Couldn't find")) {
-        updateSearchGif(<div />);
-        updateFile(request.response);
-        updatePlugin(request.response);
-      } else {
-        updateSearchGif(<div />);
-        updatePlugin(<p>Svg not found. Try searching for abstract.</p>);
+  };
+  const createIcons = (option) => {
+    const iconDivs = [];
+    const limit =
+      EOSIconsList[option].length < 10 ? EOSIconsList[option].length : 10;
+    // eslint-disable-next-line no-plusplus
+    for (let i = 0; i < limit; i++) {
+      const icon = EOSIconsList[option][i];
+      const name = EOSIcons[`Eos${icon.name}Filled`]
+        ? `Eos${icon.name}Filled`
+        : `Eos${icon.name}Outlined`;
+      const EOSReactIcon = EOSIcons[name];
+      if (EOSReactIcon !== undefined) {
+        iconDivs.push(
+          <div
+            className="image-container"
+            key={name}
+            onClick={() => {
+              copyToClipboard(EOSReactIcon({ size: "xxxl" }));
+            }}
+          >
+            {EOSReactIcon({ size: "xl" })}
+          </div>
+        );
       }
-    };
-    request.send();
+    }
+    return (
+      <div key={option}>
+        <div className="category-container">
+          <span>{option}</span>
+        </div>
+        <div className="image-box">{iconDivs}</div>
+      </div>
+    );
   };
-  const handleKeyUp = (event) => {
-    updatePlugin(<p>We would be searching for {refContainer.current.value}</p>);
-    if (event.key === "Enter") {
-      event.preventDefault();
+  useEffect(() => {
```

Comment:
One of the Rules of React Hooks is that it must be called at the top level and before any return statement.
Maybe it's a good idea to have it moved right after your `useRef` on `line 13`